libcontainer/cgroups/fs/memory.go

```diff
@@ -141,6 +141,20 @@ func setMemoryAndSwap(path string, cgroup *configs.Cgroup) error {
 			}
 		}
 	}
+	if cgroup.Resources.Memory != 0 && cgroup.Resources.MemorySwap == -1 {
+		if err := writeFile(path, "memory.limit_in_bytes", strconv.FormatInt(cgroup.Resources.Memory, 10)); err != nil {
+			return err
+		}
+		if err := writeFile(path, "memory.memsw.limit_in_bytes", strconv.FormatInt(cgroup.Resources.MemorySwap, 10)); err != nil {
+			return err
+		}
+	} else {
+		if cgroup.Resources.MemorySwap == -1 {
+			if err := writeFile(path, "memory.memsw.limit_in_bytes", strconv.FormatInt(cgroup.Resources.MemorySwap, 10)); err != nil {
```

Comment:
Why are you doing this twice? Just take this out of both if branches and add it after checking that `cgroup.Resources.Memory != 0`.